sound/soc/codecs/rt711.c

```diff
@@ -426,7 +420,7 @@ static int rt711_set_jack_detect(struct snd_soc_component *component,
 
 	/* power off */
 	regmap_write(rt711->regmap, RT711_SET_AUDIO_POWER_STATE, AC_PWRST_D3);
-	snd_soc_dapm_mutex_unlock(dapm);
+	mutex_unlock(&rt711->calibrate_mutex);
```

Comment:
the sequence where the state moves to D0 then to D3 seems odd. Shouldn't you tie into the DAPM states so that you don't power off if DAPM is asking to power up?

Reply:
Thanks. It should tie into the DAPM states here. I will update it tomorrow.